pkg/compact/compact.go

```diff
@@ -874,10 +881,25 @@ func NewBucketCompactor(
 	}, nil
 }
 
+// removeAllNonCGDirs removes all subdirectories of the given directory
+// that are not part of the given compaction groups, and all files.
+// This is needed in the case when the compaction process resumes without
+// restarting the whole process but the blocks do not exist in the remote
+// object storage anymore.
+func removeAllNonCGDirs(groups []*Group, compactDir string) error {
+	ignoreDirs := []string{}
+	for _, gr := range groups {
+		ignoreDirs = append(ignoreDirs, gr.Key())
+	}
+	return runutil.DeleteAllExceptDirs(compactDir, ignoreDirs)
+}
+
 // Compact runs compaction over bucket.
 func (c *BucketCompactor) Compact(ctx context.Context) (rerr error) {
 	defer func() {
-		if IsHaltError(rerr) {
+		// Leave the compact directory for inspection if it is a halt error
```

Comment:
this comment although appreciated, from my pov we could improve its readability.

Especially the 2nd line, has imho to much logic word. (or if it is not not, then) which maybe could simplified the sentence and improve readability imho